stdlib/src/builtin/string_literal.mojo

```diff
@@ -407,23 +408,33 @@ struct StringLiteral(
         return __mlir_op.`pop.string.size`(self.value)
 
     @always_inline("nodebug")
-    # FIXME(MSTDL-956): This should return a pointer with StaticConstantOrigin.
-    fn unsafe_ptr(self) -> UnsafePointer[UInt8]:
+    fn unsafe_ptr[
+        is_mutable: Bool = False,
+        origin: Origin[is_mutable]
+        .type = _lit_mut_cast[StaticConstantOrigin, is_mutable]
+        .result,
+    ](self) -> UnsafePointer[Byte, is_mutable=is_mutable, origin=origin]:
         """Get raw pointer to the underlying data.
 
+        Parameters:
+            is_mutable: Whether the origin is mutable.
+            origin: The origin of the data.
+
         Returns:
             The raw pointer to the data.
         """
+        constrained[not is_mutable, "StringLiteral can't be mutated"]()
```

Comment:
**Question** Given that `is_mutable` is a parameter but it's invalid for users to pass `True` here to this method, do we need the parameter at all, or should we just hard-code `is_mutable = False` in the return type, etc like you do in `unsafe_cstr_ptr()` below?

Reply:
Because otherwise the trait won't be adhered to 
```mojo
trait Stringlike(...):
    ...
    fn unsafe_ptr[
        is_mutable: Bool, origin: Origin[is_mutable].type
    ](self) -> UnsafePointer[Byte, is_mutable=is_mutable, origin=origin]:
    ...
```